d2game/d2player/game_controls.go

```diff
@@ -848,77 +860,104 @@ func (g *GameControls) Render(target d2interface.Surface) error {
-		g.nameLabel.SetText("Automap (TAB) (Mouse 3)")
+		g.nameLabel.SetText(d2tbl.TranslateString("minipanelautomap")) //"Automap" no hotkey
 		g.nameLabel.SetPosition(400, 510)
 		g.nameLabel.Render(target)
 	}
 
 	// Display message log tooltip when hovered.
 	if g.miniPanel.IsOpen() && g.actionableRegions[miniPanelMessageLog].Rect.IsInRect(mx, my) {
-		g.nameLabel.SetText("Message Log (M)")
+		g.nameLabel.SetText(d2tbl.TranslateString("minipanelmessage")) //"Message Log" no hotkey
 		g.nameLabel.SetPosition(420, 510)
 		g.nameLabel.Render(target)
 	}
 
 	// Display quest log tooltip when hovered.
 	if g.miniPanel.IsOpen() && g.actionableRegions[miniPanelQuestLog].Rect.IsInRect(mx, my) {
-		g.nameLabel.SetText("Quest Log (Q)")
+		g.nameLabel.SetText(d2tbl.TranslateString("minipanelquest")) //"Quest Log" no hotkey
 		g.nameLabel.SetPosition(440, 510)
 		g.nameLabel.Render(target)
 	}
 
 	// Display game menu tooltip when hovered.
 	if g.miniPanel.IsOpen() && g.actionableRegions[miniPanelGameMenu].Rect.IsInRect(mx, my) {
-		g.nameLabel.SetText("Game Menu (ESC)")
+		g.nameLabel.SetText(d2tbl.TranslateString("minipanelmenubtn")) //"Game Menu (Esc)" // the (Esc) is hardcoded in.
 		g.nameLabel.SetPosition(460, 510)
 		g.nameLabel.Render(target)
 	}
 
+	// Create and format Stamina string from string lookup table.
+	fmtStamina := d2tbl.TranslateString("panelstamina")
+	staminaCurr, staminaMax := int(g.hero.Stats.Stamina), int(g.hero.Stats.MaxStamina)
+	strPanelStamina := fmt.Sprintf(fmtStamina, staminaCurr, staminaMax)
+
 	// Display stamina tooltip when hovered.
 	if g.miniPanel.IsOpen() && g.actionableRegions[stamina].Rect.IsInRect(mx, my) {
-		g.nameLabel.SetText(fmt.Sprintf("Stamina: %v / %v", float64(g.hero.Stats.Stamina), float64(g.hero.Stats.MaxStamina)))
+		g.nameLabel.SetText(strPanelStamina)
 		g.nameLabel.SetPosition(320, 535)
 		g.nameLabel.Render(target)
 	}
 
+	// Display run/walk tooltip when hovered.  Note that whether the player is walking or running, the tooltip is the same in Diablo 2.
+	if g.actionableRegions[walkRun].Rect.IsInRect(mx, my) && !g.hero.IsRunToggled() {
+		g.nameLabel.SetText(d2tbl.TranslateString("RunOn")) //"Run" no hotkeys
+		g.nameLabel.SetPosition(263, 563)
+		g.nameLabel.Render(target)
+	}
+
+	if g.actionableRegions[walkRun].Rect.IsInRect(mx, my) && g.hero.IsRunToggled() {
+		g.nameLabel.SetText(d2tbl.TranslateString("RunOff")) //"Walk" no hotkeys
+		g.nameLabel.SetPosition(263, 563)
+		g.nameLabel.Render(target)
+	}
+
+	// Create and format Experience string from string lookup table.
+	fmtExp := d2tbl.TranslateString("panelexp")
+	// The English string for "panelexp" is "Experience: %u / %u", however %u doesn't translate well. So
+	// we need to rewrite %u into a formatable Go verb. %d is used in other strings, so we go with that,
+	// keeping in mind that %u likely referred to an unsigned integer.
+	fmtExp = strings.Replace(fmtExp, "%u", "%d", -1)
```

Comment:
I'm not convinced this was the best way to do this.

Reply:
I think this creates a lint error. Use `strings.ReplaceAll` instead.

Reply:
Fixed in a commit